### DISCUSSION TITLE
【求解】请问关于ethStorage的问题

### DISCUSSION THREAD

**Discussion by zedz** [IntensiveCoLearning/Web3-URL#154] [Q&A] [answered]
请问ethStorage里面的存储数据是不是只在uploader local disk-DAN-provider local disk这三个地方，没有写到链上去？存储数据是不公开的？

想要把玩家的手牌和出牌藏起来，似乎不上链的ethStorage非常适合？但是存储数据（牌）可以通过存储证明调用吗？dApp可不可以通过什么方式直接调用存储的数据（牌）？

**Comment by qzhodl** [comment]
链上存储数据的哈希值，链下在 EthStorage 的 layer2 网络里面存储数据。数据是公开的，任何人都可以通过[接口](https://github.com/ethstorage/storage-contracts-v1/blob/136d87f338a8ea00a3311dd2870fef0e3724ad1e/contracts/DecentralizedKV.sol#L170)来拿到数据

**Reply by zedz** [reply]
https://w3-drive.w3eth.io/
Your Private Hard Drive on Blockchain

这样是不是就不是公开的了？只有有密码的人才能访问到数据？

**Reply by qzhodl** [reply]
对的，w3drive是应用层做了数据加密，也就是存储在EthStorage里面的数据本身就是加密过的

**Reply by zedz** [reply]
是如何加密的？有没有范例？是可以被破解的吗？牌存在里面会不会有被偷看的风险？应用层加密的意思是前端加密过后才被存储吗？

**Reply by qzhodl** [reply, marked as answer]
意思是 w3drive 这个应用做了加密的处理，具体机制可以参考开源的repo：https://github.com/ethstorage/w3drive
